### PR TITLE
Fix RhGetCurrentThunkContext/RhCommonStub on linux-bionic-arm

### LINKED ISSUE #100191: [NATIVEAOT] lld duplicate symbol compilation error when building for linux-bionic-arm
### Description

When build a very complex library with native aot for linux-bionic-arm, `clang` fails to link with the following error;
```
ld.lld : error : duplicate symbol: RhGetCurrentThunkContext [D:\test\large_aot.csproj]
  >>> defined at PalRedhawkUnix.cpp:483 (/__w/1/s\src/coreclr/nativeaot/Runtime/unix/PalRedhawkUnix.cpp:483)
  >>>            PalRedhawkUnix.cpp.o:(RhGetCurrentThunkContext) in archive C:\nuget\packages\microsoft.netcore.app.runtime.nativeaot.linux-bionic-arm\9.0.0-preview.4.24172.18/runtimes/linux-bionic-arm/\native\libRuntime.WorkstationGC.a
  >>> defined at InteropThunksHelpers.S:54 (/__w/1/s/src/coreclr/nativeaot/Runtime/arm\InteropThunksHelpers.S:54)
  >>>            InteropThunksHelpers.S.o:(.text+0x3F) in archive C:\nuget\packages\microsoft.netcore.app.runtime.nativeaot.linux-bionic-arm\9.0.0-preview.4.24172.18/runtimes/linux-bionic-arm/\native\libRuntime.WorkstationGC.a
```

### Reproduction Steps

Unfortunately, I'm not able to create a minimal reproduction. The environment is properly set up to build for linux-bionic as both linux-bionic-arm64 and linux-bionic-x64 succeeds when building. A simple hello world sample also does not fail.

### Expected behavior

`clang` successfully compiles the native code and a native library is generated.

### Actual behavior

`clang` fails when linking.

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by filipnavara**
Fixes #100191

### CODE REVIEW

File: src/coreclr/nativeaot/Runtime/arm/InteropThunksHelpers.S [hunk 1]
```diff
@@ -46,6 +46,7 @@ LEAF_ENTRY RhGetCommonStubAddress, _TEXT
           bx            lr
 LEAF_END RhGetCommonStubAddress, _TEXT
 
+#ifndef FEATURE_EMULATED_TLS
 //
 // IntPtr RhGetCurrentThunkContext()
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/arm/InteropThunksHelpers.S:51]: Is there a reason for why this is implemented in arch specific assembly in the first place?

Comment (filipnavara) [reply src/coreclr/nativeaot/Runtime/arm/InteropThunksHelpers.S:51]: I don't think so. This just matches what other platforms were doing but I assume we can remove the assembly implementations everywhere.

Comment (jkotas) [reply src/coreclr/nativeaot/Runtime/arm/InteropThunksHelpers.S:51]: I think it would be best to delete all this unnecessary asm code.

**Review by jkotas — APPROVED** [review]
Thanks!

**Pull request merged**